lib/Locations.php

```diff
@@ -10,23 +10,31 @@
 /**
  * @implements IteratorAggregate<Location>
  */
-class Locations implements IteratorAggregate, Countable
+final class Locations implements IteratorAggregate, Countable
 {
     /**
      * @var Location[]
      */
     private $locations = [];
 
     /**
-     * @param Location[] $locations
+     * @param iterable<Location> $locations
      */
-    public function __construct(array $locations)
+    public function __construct(iterable $locations)
     {
         foreach ($locations as $location) {
             $this->add($location);
         }
     }
 
+    /**
+     * @param iterable<Location> $locations
+     */
+    public static function bySorting(iterable $locations, ?callable $sortStrategy = null): self
```

Comment:
I think I would prefer:

```
$locations = (new Locations($locations))->sorted();
```

with no `$sortStrategy` - extension points are easy to add but hard to remove, and can be implemented in many different ways. So for now I think having a single, closed, `sorted()` factory method is enough, and probably nothing more would be needed?

Reply:
Done